app/src/modules/whyis-dataset.js

```diff
@@ -0,0 +1,282 @@
+
+const defaultDataset = {
+  title: '',
+  description: '',
+  contactpoint: {
+    '@type': 'schemaPerson',
+    '@id': null,
+    name: '',
+    cpfirstname: '',
+    cplastname: '',
+    cpemail: ''
+  },
+  contributor: [],
+  author: [],
+  datepub: {
+    '@type': 'date',
+    '@value': ''
+  },
+  datemod: {
+    '@type': 'date',
+    '@value': ''
+  },
+  refby: [],
+  distribution: {
+    accessURL: null
+  },
+  depiction: {
+    name: '',
+    accessURL: null
+  }
+}
+
+const dcat = 'http://w3.org/ns/dcat#'
+const dct = 'http://purl.org/dc/terms/'
+const vcard = 'http://www.w3.org/2006/vcard/ns#'
+const foaf = 'http://xmlns.com/foaf/0.1/'
+const schema = 'http://schema.org/'
+
+const datasetFieldUris = {
+  baseSpec: 'http://semanticscience.org/resource/hasValue',
+  title: `${dct}title`,
+  description: `${dct}description`,
+
+  contactpoint: `${dcat}contactpoint`,
+  cpemail: `${vcard}email`,
+  cpfirstname: `${schema}givenName`,
+  cplastname: `${schema}familyName`,
+  individual: `${vcard}individual`,
+  schemaPerson: `${schema}Person`,
+
+  author: `${dct}creator`,
+  name: `${foaf}name`,
+  contributor: `${dct}contributor`,
+  organization: `${foaf}Organization`,
+  person: `${foaf}Person`,
+  onbehalfof: 'http://www.w3.org/ns/prov#actedOnBehalfOf',
+  specializationOf: 'http://www.w3.org/ns/prov#specializationOf',
+
+  datepub: `${dct}issued`,
+  datemod: `${dct}modified`,
+  date: 'https://www.w3.org/2001/XMLSchema#date',
+
+  refby: `${dct}isReferencedBy`,
+
+  distribution: `${dcat}distribution`,
+  depiction: `${foaf}depiction`,
+  hasContent: 'http://vocab.rpi.edu/whyis/hasContent',
+  accessURL: `${dcat}accessURL`
+}
+
+const datasetPrefix = 'dataset'
+
+// Generate a randum uuid, or use current if exists
+function generateDatasetId (guuid) {
+  var datasetId
+  if (arguments.length === 0) {
+    const { v4: uuidv4 } = require('uuid')
+    datasetId = uuidv4()
+  } else {
+    datasetId = guuid
+  }
+  return `${lodPrefix}/explorer/${datasetPrefix}/${datasetId}`
+}
+
+function buildDatasetLd (dataset) {
+  dataset = Object.assign({}, dataset)
+  dataset.context = JSON.stringify(dataset.context)
+  const datasetLd = {
+    '@id': dataset.uri,
+    '@type': []
+  }
+
+  if (dataset['@type'] != null) {
+    datasetLd['@type'].push(dataset['@type'])
+  }
+
+  Object.entries(dataset)
+    // filter out the ones that aren't in our allowed fields
+    .filter(([field, value]) => datasetFieldUris[field.toLowerCase()])
+    .forEach(([field, value]) => {
+      // make a new dictionary
+      var ldValues = {}
+      // If the field has a value
+      if (!isEmpty(value)) {
+        ldValues = recursiveFieldSetter([field, value])
+        datasetLd[datasetFieldUris[field.toLowerCase()]] = ldValues
+      }
+    })
+  return datasetLd
+}
+
+// Recursively check if a value is empty
+function isEmpty (value) {
+  // Base case
+  if ((value === '') || (value === null) || (value === []) || (value === 'undefined')) {
```

Comment:
Could we also update this to `if (![undefined, null, '', []].includes(value))`

Reply:
Oh yeah definitely, this is old migrated code.

Reply:
Ah I see